src/promptflow/promptflow/executor/_tool_invoker.py

```diff
@@ -2,9 +2,95 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
 
+from dataclasses import dataclass
+from functools import partial
+from typing import Callable, Dict
+
 from promptflow._core.tool import ToolInvoker
+from promptflow.contracts.flow import InputAssignment, Node, ToolSource
+from promptflow.exceptions import ErrorTarget
+from promptflow.executor._docstring_parser import DocstringParser
+from promptflow.executor._errors import UnsupportedAssistantToolType
+from promptflow.executor._tool_resolver import ToolResolver
 
 
 class DefaultToolInvoker(ToolInvoker):
     def invoke_tool(self, f, *args, **kwargs):
         return f(*args, **kwargs)  # Do nothing
+
+
+@dataclass
+class AssistantTool:
+    name: str
+    definition: dict
+    func: Callable
+
+
+class AssistantToolInvoker:
```

Comment:
If target to let customer use this invoker. Move it to a non-private folder in future PR.